src/sqlfluff/dialects/dialect_oracle.py

```diff
@@ -0,0 +1,90 @@
+"""The Oracle dialect.
+
+This inherits from the ansi dialect.
+"""
+from sqlfluff.core.dialects import load_raw_dialect
+from sqlfluff.core.parser.segments import NewlineSegment
+from sqlfluff.core.parser.segments.raw import KeywordSegment
+from sqlfluff.core.parser import (
+    Anything,
+    BaseSegment,
+    CommentSegment,
+    Ref,
+    Sequence,
+    StartsWith,
+    StringParser,
+    OptionallyBracketed,
+    OneOf,
+)
+
+ansi_dialect = load_raw_dialect("ansi")
+oracle_dialect = ansi_dialect.copy_as("oracle")
+
+oracle_dialect.sets("unreserved_keywords").difference_update(["COMMENT"])
+oracle_dialect.sets("reserved_keywords").update(["ON"])
```

Comment:
If removing `COMMENT` from `unreserved_keywords`, we should add it to `reserved_keywords` as well. also `ON` is already a reserved keyword in ANSI which you have inherited so don't need to add that again.

```suggestion
oracle_dialect.sets("unreserved_keywords").difference_update(["COMMENT"])
oracle_dialect.sets("reserved_keywords").update(["COMMENT"])
```